service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseDeletionControllerTest.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.OverrideAutoConfiguration;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.fpl.utils.CoreCaseDataStoreLoader.callbackRequest;
+
+@ActiveProfiles("integration-test")
+@WebMvcTest(CaseDeletionController.class)
+@OverrideAutoConfiguration(enabled = true)
+class CaseDeletionControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    private static final String AUTH_TOKEN = "Bearer token";
+    private static final String USER_ID = "1";
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Test
+    void shouldRemoveAllCaseDetailsWhenCalled() throws Exception {
```

Comment:
Correct me if i'm wrong Sam but will case data not always be empty to begin with since it was never populated with data?

Reply:
.content(MAPPER.writeValueAsString(callbackRequest())))

public static CallbackRequest callbackRequest() throws IOException {
    String response = ResourceReader.readString("core-case-data-store-api/callback-request.json");
        return mapper.readValue(response, CallbackRequest.class);
}

I use the JSON files to populate the content block in the response.

Reply:
Ah nice!